docs/Testing.md

```diff
@@ -1,13 +1,144 @@
 <h1 align="center">Fastify</h1>
 
 ## Testing
+
 Testing is one of the most important parts of developing an application. Fastify is very flexible when it comes to testing and is compatible with most testing frameworks (such as [Tap](https://www.npmjs.com/package/tap), which is used in the examples below).
 
-<a name="inject"></a>
-### Testing with http injection
+Let's `cd` into a fresh directory called 'testing-example' and type `npm init -y` in our terminal.
+
+run `npm install fastify && npm install tap pino-pretty --save-dev`
+
+### Separating concerns makes testing easy
+
+ First we're going to separate our application code from our server code:
+
+app.js
+
+```js 
+const Fastify = require('fastify')
+
+function build(opts={}) {
+  const fastify = Fastify(opts)
+  fastify.get('/', async (request, reply) => {
```

Comment:
```suggestion
  fastify.get('/', async function (request, reply) {
```

using the async function will let the user to split better the handler in smaller files since it will be able to write:

```
module.exports = async function (request, reply) {
   const user = await this.mongo....findOne()
}
```

thanks to the binding of the fastify instance.

I think this is not so publicise